src/decorators.ts

```diff
@@ -126,6 +126,32 @@ export function ResponseSchema(
         },
       }
 
+      const oldSchema =
+        source.responses[statusCode]?.content[contentType].schema
+
+      const makeNewSchema = () => {
+        if (oldSchema.oneOf) {
+          return {
+            oneOf: _.concat([...oldSchema.oneOf], schema),
+          }
+        }
+
+        return { oneOf: [{ ...oldSchema }, schema] }
+      }
+
+      if (oldSchema?.$ref || oldSchema?.items || oldSchema?.oneOf) {
+        // case where we're adding multiple schemas under single statuscode/contentType
+        const newStatusCodeResponse = _.merge(
+          {},
+          source.responses[statusCode],
+          responses[statusCode]
+        )
+        const newSchema = makeNewSchema()
+        newStatusCodeResponse.content[contentType].schema = newSchema
+        source.responses[statusCode] = newStatusCodeResponse
+        return source
```

Comment:
Is there any way we could avoid the `_.merge` here? How about just assigning to `source.responses[statusCode].content[contentType].schema = ...`? 

Also, I'm not sure if extracting `makeNewSchema` into another function provides much additional value. We can probably replace it with a simple ternary: `const newSchema = oldSchema.oneof ? ... : ...`

Reply:
> Also, I'm not sure if extracting makeNewSchema into another function provides much additional value. We can probably replace it with a simple ternary: const newSchema = oldSchema.oneof ? ... : ...

I agree

> Is there any way we could avoid the _.merge here? How about just assigning to source.responses[statusCode].content[contentType].schema = ...?

For this, my original thought with leaving the _.merge in was to keep descriptions for statusCodes. Consider the following example
```ts
@ResponseSchema(Model1)
@ResponseSchema(Model2, {description: xxx})
```

If we drop merge and just set schema, then I *think* that the first description would be lost. Lmk what you think.

Reply:
Right, better leave the merge in there for now then.